src/uwtools/drivers/forecast.py

```diff
@@ -162,12 +162,28 @@ def job_card(self):
         Turns the resources config object into a batch card for the configured Task.
         """
 
-    def run(self):
-        """
-        Runs the forecast executable with the namelist file and staged input files.
-
-        This will both build the executable and run it.
-        """
+    def run(
+        self,
+        exec_name: str,
+        export: Optional[str] = None,
+        mpi: Optional[str] = None,
+        node_arg: Optional[str] = None,
+        pe_member01: Optional[int] = None,
+        num_of_procs: Optional[int] = None,
+        oversubscribe: Optional[str] = None,
+        procs_per_nodes: Optional[str] = None,
+        ppn: Optional[int] = None,
+        cpu_bind: Optional[str] = None,
+        core: Optional[str] = None,
+        depth: Optional[str] = None,
+        omp_num_threads: Optional[int] = None,
+    ) -> str:
+        """
+        Builds the forecast executable with the user provided run commands.
+
+        This will build the executable.
+        """
+        return (" ".join(str(i) if i is not None else "" for i in vars().values())).strip()
```

Comment:
A quick search didn't turn up for me any docs saying that `vars()` *must* be ordered the same way as the arguments are given in the function, but it seems to be the case and is probably reliable.

Note that `vars()` will also pick up any locally-defined variables:
```
>>> def foo(a, b=None, c=None):
...   d = 88
...   return (" ".join(str(i).strip() for i in filter(None, vars().values())))
... 
>>> foo("hello", c="world")
'hello world 88'
```
So beware, but I don't think we need to change anything now.

Reply:
That comment was just musing about possible issues that don't seem to currently be in play, so I'll resolve this conversation to keep this PR tidy.